### PR TITLE
fix progress bar restart with fault-tolerant training enabled

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Fixes a TODO in #9130 

The progress bar callback did not resume correctly, because of two reasons:
- `on_train_start()` was resuming the correct progress value but `on_train_epoch_start()` would overwrite it back to 0
- the `on_*_epoch_start` hooks reset the tqdm counter to 0 instead of the value we would resume from

This came up in the context of fault tolerant training. 

<details>
  <summary>Repro script</summary>

```python
import os
from time import sleep

import torch
from torch.utils.data import DataLoader, Dataset

from pytorch_lightning import LightningModule, Trainer


class RandomDataset(Dataset):
    def __init__(self, size, length):
        self.len = length
        self.data = torch.randn(length, size)

    def __getitem__(self, index):
        return self.data[index]

    def __len__(self):
        return self.len


class BoringModel(LightningModule):
    def __init__(self):
        super().__init__()
        self.layer = torch.nn.Linear(32, 2)

    def forward(self, x):
        return self.layer(x)

    def training_step(self, batch, batch_idx):
        sleep(0.5)
        print(batch_idx, self.trainer.fit_loop.batch_idx)
        loss = self(batch).sum()
        self.log("train_loss", loss)
        return {"loss": loss}

    def configure_optimizers(self):
        return torch.optim.SGD(self.layer.parameters(), lr=0.1)


def run():
    train_data = DataLoader(RandomDataset(32, 64), batch_size=2)

    model = BoringModel()
    trainer = Trainer(
        default_root_dir=os.getcwd(),
        weights_summary=None,
        # resume_from_checkpoint=".pl_auto_save.ckpt",
    )
    trainer.fit(model, train_dataloaders=train_data)


if __name__ == "__main__":
    run()

```

</details>

**The added test fails on master.**

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/trainer/trainer.py [hunk 1]
```diff
@@ -511,6 +511,7 @@ def _call_and_handle_interrupt(self, trainer_fn: Callable, *args: Any, **kwargs:
             if not self.interrupted:
                 self.state.status = TrainerStatus.INTERRUPTED
                 self.on_keyboard_interrupt()
+                self._on_exception()
```
Comment (awaelchli) [inline pytorch_lightning/trainer/trainer.py:514]:
```suggestion
```

**Review by carmocca — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
Neat !

**Review by Borda — APPROVED** [review]
